src/internal/components/button-trigger/index.tsx

```diff
@@ -74,22 +74,28 @@ const ButtonTrigger = (
       inFilteringToken && styles['in-filtering-token'],
       inlineTokens && styles['inline-tokens']
     ),
-    disabled: disabled || readOnly,
+    disabled: disabled,
     'aria-expanded': pressed,
     'aria-label': ariaLabel,
     'aria-labelledby': ariaLabelledby,
     'aria-describedby': ariaDescribedby,
     'aria-haspopup': ariaHasPopup ?? 'listbox',
     'aria-controls': ariaControls,
-    onKeyDown: onKeyDown && (event => fireKeyboardEvent(onKeyDown, event)),
-    onKeyUp: onKeyUp && (event => fireKeyboardEvent(onKeyUp, event)),
-    onMouseDown: onMouseDown && (event => fireCancelableEvent(onMouseDown, {}, event)),
-    onClick: onClick && (event => fireCancelableEvent(onClick, {}, event)),
-    onFocus: onFocus && (event => fireCancelableEvent(onFocus, {}, event)),
-    onBlur: onBlur && (event => fireCancelableEvent(onBlur, { relatedTarget: event.relatedTarget }, event)),
     autoFocus,
   };
 
+  if (!readOnly) {
+    attributes = {
+      ...attributes,
+      onKeyDown: onKeyDown && (event => fireKeyboardEvent(onKeyDown, event)),
+      onKeyUp: onKeyUp && (event => fireKeyboardEvent(onKeyUp, event)),
+      onMouseDown: onMouseDown && (event => fireCancelableEvent(onMouseDown, {}, event)),
+      onClick: onClick && (event => fireCancelableEvent(onClick, {}, event)),
+      onFocus: onFocus && (event => fireCancelableEvent(onFocus, {}, event)),
+      onBlur: onBlur && (event => fireCancelableEvent(onBlur, { relatedTarget: event.relatedTarget }, event)),
```

Comment:
From the api documents readOnly controls are still focusable, should these events then still be fired?

Reply:
No, they should not. The element's focus is managed by css, not by these events. They can cause unexpected behavior for `readOnly` state